src/main.rs

```diff
@@ -17,110 +17,76 @@
 
 mod info;
 use info::info;
-pub mod manifest_config;
-use manifest_config::ManifestConfig;
-mod fs_report;
 mod signer;
-use signer::get_c2pa_signer;
+use signer::SignConfig;
 
-// define the command line options
-#[derive(Debug, StructOpt)]
-#[structopt(about = "Tool for displaying and creating C2PA manifests.",global_settings = &[AppSettings::ColoredHelp, AppSettings::ArgRequiredElseHelp])]
+/// Tool for displaying and creating C2PA manifests."
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None, setting = AppSettings::ArgRequiredElseHelp)]
 struct CliArgs {
-    #[structopt(parse(from_os_str))]
-    #[structopt(
-        short = "m",
-        long = "manifest",
-        help = "Path to manifest definition JSON file."
-    )]
+    #[clap(short, long, help = "Path to manifest definition JSON file.")]
     manifest: Option<PathBuf>,
 
-    #[structopt(parse(from_os_str))]
-    #[structopt(short = "o", long = "output", help = "Path to output file or folder.")]
+    #[clap(short, long, help = "Path to output file or folder.")]
     output: Option<PathBuf>,
 
-    #[structopt(parse(from_os_str))]
-    #[structopt(short = "p", long = "parent", help = "Path to a parent file.")]
+    #[clap(short, long, help = "Path to a parent file.")]
     parent: Option<PathBuf>,
 
-    #[structopt(
-        short = "c",
-        long = "config",
-        help = "Manifest definition passed as a JSON string."
-    )]
+    #[clap(short, long, help = "Manifest definition passed as a JSON string.")]
     config: Option<String>,
 
-    #[structopt(
-        short = "d",
-        long = "detailed",
-        help = "Display detailed C2PA-formatted manifest data."
-    )]
+    #[clap(short, long, help = "Display detailed C2PA-formatted manifest data.")]
     detailed: bool,
 
-    #[structopt(
-        short = "f",
-        long = "force",
-        help = "Force overwrite of output if it already exists."
-    )]
+    #[clap(short, long, help = "Force overwrite of output if it already exists.")]
     force: bool,
 
-    /// The path to an asset to examine or embed a manifest into.
-    #[structopt(parse(from_os_str))]
+    #[clap(help = "The path to an asset to examine or embed a manifest into.")]
     path: PathBuf,
 
-    #[structopt(
-        short = "r",
-        long = "remote",
-        help = "Embed remote URL manifest reference."
-    )]
+    #[clap(short, long, help = "Embed remote URL manifest reference.")]
     remote: Option<String>,
 
-    #[structopt(
-        short = "s",
-        long = "sidecar",
-        help = "Generate a sidecar (.c2pa) manifest"
-    )]
+    #[clap(short, long, help = "Generate a sidecar (.c2pa) manifest")]
     sidecar: bool,
 
-    #[structopt(long = "tree", help = "Create a tree diagram of the manifest store.")]
+    #[clap(short, long, help = "Write ingredient report and assets to a folder.")]
+    ingredient: bool,
+
+    #[clap(long, help = "Create a tree diagram of the manifest store.")]
```

Comment:
Clap should parse the `help` attribute automatically if it's formatted as a comment. Not an issue here because the help args are short, but it may be useful in the future.